<!doctype html>
<html lang="en">

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>Async/Await</title>
  <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0-beta1/dist/css/bootstrap.min.css" rel="stylesheet"
    integrity="sha384-0evHe/X+R7YkIZDRvuzKMRqM+OrBnVFBL6DOitfPri4tjfHxaWutUpFmBp4vmVor" crossorigin="anonymous">
</head>

<body>
  <div class="container">
    <center>
      <h2>Async/Await </h2>
    </center>
  </div>
  <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0-beta1/dist/js/bootstrap.bundle.min.js"
    integrity="sha384-pprn3073KE6tl6bjs2QrFaJGz5/SUsLqktiwsUTF55Jfv3qYSDhgCecCxMW52nD2"
    crossorigin="anonymous"></script>
</body>

<script>
  async function func1() {  //  ye ek promise return kr dega 
    console.log('Inside function');
    const responce = await fetch('https://api.github.com/users')  //  api link
    console.log('Before responce')
    const user = await responce.json()
    return user;
  }

  console.log('Before calling function');
  let a = func1();
  console.log('After calling function');
  a.then(data => console.log(data));
  console.log(a);
  console.log('Last line of this file');  //  sub run hone ke bad await wala sentencr run hoga yani ki fatch hoga api
</script>

</html>